src/overloads/arrays.jl

```diff
@@ -0,0 +1,231 @@
+    return primals, tracers
+end
+function split_dual_array(A::SparseArrays.SparseMatrixCSC{D}) where {D<:Dual}
+    A = Matrix(A)
+    primals = getproperty.(A, :primal)
+    tracers = getproperty.(A, :tracer)
+    return sparse(primals), sparse(tracers)
+end
+
+#==================#
+# LinearAlgebra.jl #
+#==================#
+
+# TODO: replace `second_order_or` by less conservative sparsity patterns when possible
+
+## Determinant
+LinearAlgebra.det(A::AbstractMatrix{T}) where {T<:AbstractTracer} = second_order_or(A)
+LinearAlgebra.logdet(A::AbstractMatrix{T}) where {T<:AbstractTracer} = second_order_or(A)
+function LinearAlgebra.logabsdet(A::AbstractMatrix{T}) where {T<:AbstractTracer}
+    t1 = second_order_or(A)
+    t2 = sign(t1) # corresponds to sign of det(A): set first- and second-order derivatives to zero
+    return (t1, t2)
+end
+
+## Norm
+function LinearAlgebra.norm(A::AbstractArray{T}, p::Real=2) where {T<:AbstractTracer}
+    return second_order_or(A)
+end
+function LinearAlgebra.opnorm(A::AbstractArray{T}, p::Real=2) where {T<:AbstractTracer}
+    if isone(p)
+        return first_order_or(A)
+    else
+        return second_order_or(A)
+    end
+end
+function LinearAlgebra.opnorm(A::AbstractMatrix{T}, p::Real=2) where {T<:AbstractTracer}
+    if isone(p)
+        return first_order_or(A)
+    else
+        return second_order_or(A)
+    end
+end
+
+## Eigenvalues
+
+function LinearAlgebra.eigmax(
+    A::Union{T,AbstractMatrix{T}}; permute::Bool=true, scale::Bool=true
+) where {T<:AbstractTracer}
+    return second_order_or(A)
+end
+function LinearAlgebra.eigmin(
+    A::Union{T,AbstractMatrix{T}}; permute::Bool=true, scale::Bool=true
+) where {T<:AbstractTracer}
+    return second_order_or(A)
+end
+function LinearAlgebra.eigen(
+    A::AbstractMatrix{T};
+    permute::Bool=true,
+    scale::Bool=true,
+    sortby::Union{Function,Nothing}=nothing,
+) where {T<:AbstractTracer}
+    LinearAlgebra.checksquare(A)
+    n = size(A, 1)
+    t = second_order_or(A)
+    values = Fill(t, n)
+    vectors = Fill(t, n, n)
+    return LinearAlgebra.Eigen(values, vectors)
+end
+
+## Inverse
+function LinearAlgebra.inv(A::StridedMatrix{T}) where {T<:AbstractTracer}
+    LinearAlgebra.checksquare(A)
+    t = second_order_or(A)
+    return Fill(t, size(A)...)
+end
+function LinearAlgebra.pinv(
+    A::AbstractMatrix{T}; atol::Real=0.0, rtol::Real=0.0
+) where {T<:AbstractTracer}
+    n, m = size(A)
+    t = second_order_or(A)
+    return Fill(t, m, n)
+end
+
+## Division
+function LinearAlgebra.:\(
+    A::AbstractMatrix{T}, B::AbstractVecOrMat
+) where {T<:AbstractTracer}
+    Ainv = LinearAlgebra.pinv(A)
+    return Ainv * B
+end
+
+## Exponent
+function LinearAlgebra.exp(A::AbstractMatrix{T}) where {T<:AbstractTracer}
+    LinearAlgebra.checksquare(A)
+    n = size(A, 1)
+    t = second_order_or(A)
+    return Fill(t, n, n)
+end
+
+## Matrix power
+function LinearAlgebra.:^(A::AbstractMatrix{T}, p::Integer) where {T<:AbstractTracer}
```

Comment:
Is it a good idea to make this type-unstable? I don't know whether it matters, and I know I'm the one who asked for special cases

Reply:
We could remove the `isone` case.
I highly doubt it is relevant in the first place.

Reply:
I could see the case where people do something like `sum(A^k for k in 1:n)` but sure, let's remove it for now

Reply:
Well, in that very specific case, were also more efficient now, unless `n=1`. ;)